game-core/src/main/java/games/strategy/engine/framework/GameDataManager.java

```diff
@@ -79,14 +76,6 @@ public static GameData loadGame(final InputStream is) throws IOException {
       final Version readVersion = (Version) input.readObject();
       final boolean headless = HeadlessGameServer.headless();
       if (!GameEngineVersion.of(ClientContext.engineVersion()).isCompatibleWithEngineVersion(readVersion)) {
-        // a hack for now, but a headless server should not try to open any savegame that is not its version
-        if (headless) {
-          final String message = "Incompatible game save, we are: " + ClientContext.engineVersion()
-              + "  Trying to load game created with: " + readVersion;
-          HeadlessGameServer.sendChat(message);
-          log.log(Level.SEVERE, message);
-          return null;
```

Comment:
I confirmed that throwing an exception here has the same effect as returning `null` in a headless game server.